src/common/com/intellij/plugins/haxe/lang/lexer/HaxeConditionalCompilationLexerSupport.java

```diff
@@ -35,8 +35,7 @@
  * Created by ebishton on 3/23/17.
  */
 public class HaxeConditionalCompilationLexerSupport {
-  static final String classname = new Object(){}.getClass().getEnclosingClass().getName();
-  static final HaxeDebugLogger LOG = HaxeDebugLogger.getInstance("#" + classname);
+  static final HaxeDebugLogger LOG = HaxeDebugLogger.getLogger(HaxeConditionalCompilationLexerSupport.class);
```

Comment:
If you don't want the leading '#', then you don't need to specify the current class, either.  You can just call HaxeDebugLogger.getLogger().  It knows how to get the calling class name.

I forget what problem I was solving when I wrote that `new Object(){}.getClass().getEnclosingClass().getName();`.  It was a very real problem, though. I guess I was looking for a method to get the logger name without having to hard code the class name.  And, I wanted the '#' for a reason, too, but I don't recall it at the moment.